DirectoryNotEmptyException in remove() on network file system

I sometimes get the following exception when running N5 tests on my workstation:
```
Testing lz4 uint8
java.nio.file.DirectoryNotEmptyException: /groups/saalfeld/home/pisarevi/tmp/n5-test/test/group/dataset/0/0
	at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:242)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.janelia.saalfeldlab.n5.N5FSWriter.lambda$remove$0(N5FSWriter.java:159)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:390)
	at java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.janelia.saalfeldlab.n5.N5FSWriter.remove(N5FSWriter.java:150)
	at org.janelia.saalfeldlab.n5.AbstractN5Test.testWriteReadByteBlock(AbstractN5Test.java:171)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
[ERROR] Tests run: 16, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 8.308 s <<< FAILURE! - in org.janelia.saalfeldlab.n5.N5FSTest
[ERROR] testWriteReadByteBlock(org.janelia.saalfeldlab.n5.N5FSTest)  Time elapsed: 0.455 s  <<< FAILURE!
java.lang.AssertionError
```

It's quite rare and most of the time the tests succeed, but it's more likely to occur in n5-spark tests, and the error is identical: https://github.com/saalfeldlab/n5-spark/issues/13

It seems that it happens only on the network file system. I've never seen this when building on a laptop where test files are written and deleted from a local drive.

--- a/src/main/java/org/janelia/saalfeldlab/n5/N5FSWriter.java
+++ b/src/main/java/org/janelia/saalfeldlab/n5/N5FSWriter.java
@@ -148,21 +148,13 @@ public class N5FSWriter extends N5FSReader implements N5Writer {
 			try (final Stream<Path> pathStream = Files.walk(path)) {
 				pathStream.sorted(Comparator.reverseOrder()).forEach(
 						childPath -> {
-							if (Files.isRegularFile(childPath)) {
-								try (final LockedFileChannel channel = LockedFileChannel.openForWriting(childPath)) {
-									Files.delete(childPath);
-								} catch (final IOException e) {
-									e.printStackTrace();
-								}
-							} else
-								try {
-									Files.delete(childPath);
-								} catch (final IOException e) {
-									e.printStackTrace();
-								}
+							try {
+								Files.delete(childPath);
+							} catch (final IOException e) {
+								e.printStackTrace();
+							}
 						});
 			}
-
 		return !Files.exists(path);
 	}
 

